### PR TITLE
team-level std algos: part 11

### PR DESCRIPTION
**Pull request by fnrizzi**
Co-authored-by: Cezary Skrzyński
Co-authored-by: Jakub Strzebonski

## Content 

Team-level implementation and tests for:

- `Kokkos_Transform.hpp` 
- `Kokkos_Generate.hpp` 
- `Kokkos_GenerateN.hpp`
- `Kokkos_SwapRanges.hpp`

## Technical comments

- the change `class` -> `typename` and the enable if defaulted in the template parameters comes from [this request](https://github.com/kokkos/kokkos/pull/6200#issuecomment-1626406816)

- the code duplication is intentional and agreed upon after the long discussion in [here](https://github.com/kokkos/kokkos/pull/6200)

- tests are run for both contiguous views and strided views

## Other comments

- all team-level PRs are tracked in: https://github.com/kokkos/kokkos/issues/6352

- this comes from breaking apart https://github.com/kokkos/kokkos/pull/5150

## Possible conflicts

Implementation-wise, this has not conflict with other team-level PRs so it could be merged independently. 
However, a trivial conflict in the CMakeLists inside the unit test occurs if another of the team-level PRs is merged before this because each team-level PR adds test using a specific "label". Resolving this is trivial. 

## Associated wiki PRs

- `Kokkos_Transform.hpp`, `Kokkos_Generate.hpp`: see [wiki PR](https://github.com/kokkos/kokkos-core-wiki/pull/386)
- `Kokkos_GenerateN.hpp`, `Kokkos_SwapRanges.hpp`: : see [wiki PR](https://github.com/kokkos/kokkos-core-wiki/pull/393)

IMPORTANT: wiki PRs are not complete yet but 99% there 

The issue tracking all *wiki PRs* is [here](https://github.com/kokkos/kokkos-core-wiki/issues/379)

### CODE REVIEW

File: algorithms/unit_tests/TestStdAlgorithmsTeamTransformUnaryOp.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  run_all_scenarios<DynamicTag, double>();
+  run_all_scenarios<StridedTwoRowsTag, int>();
+  run_all_scenarios<StridedThreeRowsTag, unsigned>();
+}
+
+// TEST(std_algorithms_transform_team_test, test_binary_op) {
+//   run_all_scenarios<DynamicTag, double>(1);
+//   run_all_scenarios<StridedTwoRowsTag, int>(1);
+//   run_all_scenarios<StridedThreeRowsTag, unsigned>(1);
+// }
```
Comment (masterleinad) [inline algorithms/unit_tests/TestStdAlgorithmsTeamTransformUnaryOp.cpp:178]: Why is this commented?

Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsTeamTransformUnaryOp.cpp:178]: my bad for not removing it, i just did. Originally the test was called `TestStdAlgorithmsTeamTransform.cpp` and contained both the test for the unary and binary op. Then i decided to split it so now there are two files , one testing the unary and one testing the binary op. I forgot to remove this code above thanks for pointing it out

**Review by masterleinad — APPROVED** [review]

**Comment by fnrizzi** [comment]
retest this please

**Review by crtrott — APPROVED** [review]

**Comment by crtrott** [comment]
Only HIP times out, but after the algorithm tests passed, and those tests weren't excessive long.

**Pull request merged**